game/scripts/npc/items/item_charge_bkb.txt

```diff
@@ -49,42 +50,138 @@
 		"ItemAliases"					"charge bkb"
 		"ItemRequiresCharges"			"1"
 		"ItemDisplayCharges"			"1"
-
-		// Special	
-		//-------------------------------------------------------------------------------------------------------------
+		"ItemPermanent"					"1"
+		"ItemInitialCharges"			"4"			//RamonNZ: This should possibly be 0 when released, but having some initial charges is nice for testing purposes atm.
 		"AbilitySpecial"
 		{
 			"01"
 			{
 				"var_type"				"FIELD_INTEGER"
-				"max_charges"			"17"
+				"bonus_strength"		"14"		//RamonNZ: BKB + added wand +4
 			}
 			"02"
 			{
 				"var_type"				"FIELD_INTEGER"
-				"charge_radius"			"1200"
+				"bonus_damage"			"24"		
 			}
 			"03"
 			{
 				"var_type"				"FIELD_INTEGER"
-				"bonus_all_stats"		"4"
+				"duration"				"0.1"		//RamonNZ: Leave as a placeholder please, and do not edit. This duration key is overwritten based on number of charges, but the lua script doesn't seem to be able to lessen the duration if you make it high.
 			}
-			// BKB
-
 			"04"
 			{
 				"var_type"				"FIELD_INTEGER"
-				"bonus_strength"		"10"
+				"max_charges"			"17"		//RamonNZ: Based on wand, seems fair, not that you'll realistically get much more than 5 or so with the charge decay.
 			}
 			"05"
 			{
 				"var_type"				"FIELD_INTEGER"
-				"bonus_damage"			"24"
+				"charge_radius"			"1200"		//RamonNZ: Wand gain charge radius.
 			}
 			"06"
 			{
 				"var_type"				"FIELD_INTEGER"
-				"model_scale"			"30"     // Percentage over model scale
+				"bonus_all_stats"		"4"			//RamonNZ: Because its combined with a wand
+			}
+			"07"
+			{
+				"var_type"				"FIELD_INTEGER"
+				"restore_per_charge"	"15"		//RamonNZ: If used - keeping the functionality of wand restore as well, because it makes sense. Set to 0 if not wanted.
+			}
+			"08"
+			{
+				"var_type"				"FIELD_INTEGER"
+				"charge_decay_time"		"20"		//RamonNZ: New KV - Charges will decay by -1 at this time interval.
+			}
+			"09"
+			{
+				"var_type"				"FIELD_INTEGER"
+				"immunity_time_per_charge"	"2"		//RamonNZ: New KV - How many seconds of spell immunity you get per charge when activated.
 			}
 		}
-	}
\ No newline at end of file
+		"OnSpellStart"
+		{
+			"RunScript"
+			{
+				"ScriptFile"			"items/charge_bkb.lua"

```

Comment:
You're missing this file, you only have `game/scripts/npc/items/item_charge_bkb.txt` in this PR